.github/workflows/promote_release.yaml

```diff
@@ -125,6 +125,24 @@ jobs:
           # Get PR's view of where main ought to be.  This is two commits behind PR's head.
           echo "PR_MAIN_REF=$(git rev-list -n 1 origin/${PR_BRANCH}^^)" >> $GITHUB_ENV
 
+      - name: 'Get run id of stage release'
+        run: |
+          STAGE_RUN_ID=$(gh issue view ${{ github.event.issue.number }} --json labels | jq  --raw-output --exit-status '[.labels.[] | select(.name | startswith("__run_id-"))][0] | .name | sub("__run_id-"; "")')
```

Comment:
I'm tagging the release PR with the run id of the workflow that ran stage.  I am using that to retrieve a JSON object contain amongst other things, the Sonar deploymentId.  That will allow the stage phase to call the CPP API to promote the release to Maven Central, or drop it.